src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java

```diff
@@ -287,10 +287,30 @@ public void hideDetails(Submission submission, User user) {
      */
     public Result setNewResult(Submission submission) {
         Result result = new Result();
+        result.setParticipation(submission.getParticipation());
+        result = resultRepository.save(result);
         result.setSubmission(submission);
         submission.setResult(result);
-        result.setParticipation(submission.getParticipation());
+        submissionRepository.save(submission);
+        return result;
+    }
+
+    /**
+     * todo wip: currently used to fix issues with saving / assigning results to submissions
```

Comment:
can you add a better documentation here instead?
also make sure to write `TODO` Nicolas Ruscher or another name

Reply:
adjusted in 00b31df355a6a714b5dc08d142c364095edabb35